### ISSUE TITLE
Ability to specify interaction path to be a regex or support variables

### ISSUE THREAD

**Issue by mboudreau** [bethesque/pact-mock_service#11]
Since we do not want to setup each restful path and then test every single one for the same interactions, the ability to specify a generic path that works for all the same paths would be the best way to move forward.

This and the ability to test data schema would solve 90% of our use cases: https://github.com/bethesque/pact-mock_service/issues/10

**Comment by bethesque** [comment]
I did this the other day, and tagged you in the issue, didn't you see it?

**Comment by bethesque** [comment]
Hm, wrong person, sorry. See this issue: https://github.com/DiUS/pact-consumer-js-dsl/issues/25

**Comment by bethesque** [comment]
Did that work?

**Comment by bethesque** [comment]
Hey @mboudreau, did you get this working? I will close if so.

**Comment by mboudreau** [comment]
@bethesque I actually haven't had the time just yet to test it out.  Will try to get it going today if possible.

**Comment by mboudreau** [comment]
Yep, works with pact-mock-service 0.2.4, except the function is as following:

```
function term(matcher, generate) {
	if(!matcher || !generate) {
		throw 'Matcher and Generate arguments must be specified to use Term';
	}
	return {
		"json_class": "Pact::Term",
		"data": {
			"generate": generate,
			"matcher": {
				"json_class": "Regexp",
				"o": 0,
				"s": matcher
			}
		}
	};
}
```

with an example usage:

```
provider
	.given('the form id based url')
	.uponReceiving('a GET request for that url')
	.withRequest('get',  term('/[a-zA-Z0-9]+', '/DSye5tXb'))
	.willRespondWith({
		status: 200,
		headers: {'Content-Type': 'application/json; charset=utf-8'},
		body: {id:'DSye5tXb'}
	});
```

**Comment by bethesque** [comment]
Cool. We can put that in the DSL once we've finished the current PR.

**Comment by kaiserk** [comment]
Hi, A year too late in the party. Any update on this one? Just created an issue in `pact-consumer-js-dsl`, https://github.com/DiUS/pact-consumer-js-dsl/issues/57 then noticed this ticket. I don't see any PR open, I can assume the PR has been merged already?

**Comment by YOU54F** [comment]
this is complete
